src/main/java/me/jellysquid/mods/sodium/client/render/pipeline/BlockRenderer.java

```diff
@@ -62,13 +67,58 @@ public BlockRenderer(LightPipelineProvider lighters) {
         this.occlusionCache = new BlockOcclusionCache();
     }
 
+    public boolean renderFluidLogged(Fluid fluid, RenderBlocks renderBlocks, BlockPos pos, ChunkModelBuffers buffers, long seed) {
+        if (fluid == null) {
+            return false;
+        }
+        Block block = fluid.getBlock();
+        if (block == null) {
+            return false;
+        }
+
+        boolean rendered = false;
+
+        try {
+            final LightMode mode = LightMode.SMOOTH; // TODO: this.getLightingMode(block); is what was previously used. The flat pipeline is busted and was only an optimization for very few blocks.
+            final LightPipeline lighter = this.lighters.getLighter(mode);
+
+            TessellatorManager.startCapturing();
+            final CapturingTessellator tess = (CapturingTessellator) TessellatorManager.get();
+            tess.startDrawingQuads();
+            // Use setTranslation rather than setOffset so that the float data written to the internal buffer
+            // is done in subchunk-relative coordinates
+            tess.setOffset(POS_ZERO);
+            tess.setTranslation(-pos.x, -pos.y, -pos.z);
+            renderBlocks.renderBlockByRenderType(block, pos.x, pos.y, pos.z);
+            final List<QuadView> quads = TessellatorManager.stopCapturingToPooledQuads();
+            tess.resetOffset();
+
+            for (ModelQuadFacing facing : ModelQuadFacing.VALUES) {
+                this.random.setSeed(seed);
+                this.renderQuadList(pos, lighter, buffers, quads, facing, (this.useAmbientOcclusion && this.useSodiumAO));
+            }
+
+            if (!quads.isEmpty()) rendered = true;
+        } finally {
+            TessellatorManager.cleanup();
+        }
+
+        return rendered;
+
+    }
+
     public boolean renderModel(IBlockAccess world, RenderBlocks renderBlocks, Block block, int meta, BlockPos pos, ChunkModelBuffers buffers, boolean cull, long seed) {
         final LightMode mode = LightMode.SMOOTH; // TODO: this.getLightingMode(block); is what was previously used. The flat pipeline is busted and was only an optimization for very few blocks.
         final LightPipeline lighter = this.lighters.getLighter(mode);
 
+        boolean rendered = false;
+
+//        if (ModStatus.isFluidLoggedLoaded) {
+//            rendered = renderFluidLogged(world, renderBlocks, pos, buffers, seed);
```

Comment:
?

Reply:
Yeah that's leftover I moved it elsewhere, need to clean that up.